### PR TITLE
Added validation for opaque int resources.

### PR DESCRIPTION
**Pull request by ConnorDoyle**
- Added validation for opaque int resources.
- Ensure supplied opaque int quantities in node capacity, node allocatable,
  pod request and pod limit are integers.
- Added tests for new validation logic (node update and pod spec).

Testing: `make check WHAT=pkg/api/validation GOFLAGS="-v -run TestValidate"`

### CODE REVIEW

**Comment by ConnorDoyle** [comment]
@balajismaniam rebased on target

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -2688,6 +2692,18 @@ func ValidateNodeUpdate(node, oldNode *api.Node) field.ErrorList {
 	return allErrs
 }
 
+// Validate that opaque integer resource quantities in a resource list are
+// indeed integers.
+func ValidateOpaqueIntQuants(rs api.ResourceList, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for rName, rQuant := range rs {
+		if strings.HasPrefix(string(rName), api.ResourceOpaqueIntPrefix) && rQuant.MilliValue() != 1000*rQuant.Value() {
+			allErrs = append(allErrs, field.Invalid(fldPath, rQuant.String(), "franctional values are not allowed for opaque integer resources"))
```
Comment (balajismaniam) [inline pkg/api/validation/validation.go]: s/franctional/fractional/

Comment (ConnorDoyle) [reply pkg/api/validation/validation.go]: Oops, good catch

Comment (ConnorDoyle) [reply pkg/api/validation/validation.go]: Fixed

**Comment by ConnorDoyle** [comment]
@balajismaniam added that missing case we discussed. Think this is ready to roll.

**Comment by balajismaniam** [comment]
Thanks for the updates. LGTM.

**Pull request merged**